### PR TITLE
HHH-13204 Introduce a configuration flag to skip processing of XML mapping metadata

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HHH-13204

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java [hunk 1]
```diff
@@ -627,6 +628,12 @@ public MetadataBuildingOptionsImpl(StandardServiceRegistry serviceRegistry) {
 
 			this.multiTenancyStrategy =  MultiTenancyStrategy.determineMultiTenancyStrategy( configService.getSettings() );
 
+			this.xmlMetadataEnabled =  configService.getSetting(
+					AvailableSettings.XML_MAPPING_ENABLED,
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java]: Minor nitpick: could we be consistent and either choose metadata or mapping? The attribute is `xmlMetadataEnabled` whereas the setting is `XML_MAPPING_ENABLED`.

Comment (Sanne) [reply hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java]: ok. which one do you want?

Comment (gsmet) [reply hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java]: I don't really care. Mapping makes more sense IMHO but it's the one requiring the most changes so not sure it's the right choice :).

**Comment by Sanne** [comment]
> Shouldn't we add a test to at least basically test this code path?

I'll try, now that I've finished the other urgent things ;)

I don't believe it's very important though. Many existing tests are covering the most important case: of old style mapping still working. If the "disabling" doesn't work, it's just slower - I tested it empirically, adding a test would somewhat protect from regressions but won't protect it to "stay fast"

**Comment by gsmet** [comment]
Yeah, it was more to test that setting it to false doesn't throw an exception somewhere in the bootstrap.

**Comment by Sanne** [comment]
pushed a fresh copy

**Comment by Sanne** [comment]
@gsmet ok to merge?

**Comment by gsmet** [comment]
Adjusted a few minor details and merged. Thanks.

**Comment by Sanne** [comment]
@gsmet your change of comments is confusing.

Read this:
 > Allows to skip processing of XML Mapping. [...] By default it's enabled.

When I originally stated "By default, the XML mapping is taken into account." it's meaning the opposite.

**Pull request closed without merging**